browser/ui/views/tabs/brave_new_tab_button.cc

```diff
@@ -74,144 +59,40 @@ SkPath BraveNewTabButton::GetBorderPath(const gfx::Point& origin,
-  // TabStripRegionView::UpdateNewTabButtonBorder() gives this button's inset.
-  // So, adding more insets here is easy solution.
-  return NewTabButton::GetInsets() + gfx::Insets::TLBR(0, 6, 0, 0);
+  NewTabButton::PaintIcon(canvas);
 }
 
 void BraveNewTabButton::PaintFill(gfx::Canvas* canvas) const {
-  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs)) {
-    NewTabButton::PaintFill(canvas);
-    return;
-  }
-
-  if (!tabs::utils::ShouldShowVerticalTabs(tab_strip_->GetBrowser())) {
-    NewTabButton::PaintFill(canvas);
-    return;
-  }
-
-  if (tab_strip_->GetCustomBackgroundId(BrowserFrameActiveState::kUseCurrent)
-          .has_value()) {
-    NewTabButton::PaintFill(canvas);
-    return;
-  }
-
-  auto* cp = GetColorProvider();
-  DCHECK(cp);
-
-  // Override fill color
-  {
-    gfx::ScopedCanvas scoped_canvas(canvas);
-    canvas->UndoDeviceScaleFactor();
-    cc::PaintFlags flags;
-    flags.setAntiAlias(true);
-    flags.setColor(cp->GetColor(kColorToolbar));
-    canvas->DrawPath(GetBorderPath(gfx::Point(), canvas->image_scale(), false),
-                     flags);
-  }
-
-  // Draw split line on the top
-  gfx::Rect separator_bounds = GetLocalBounds();
-  separator_bounds.set_height(1);
-  cc::PaintFlags flags;
-  flags.setStyle(cc::PaintFlags::kFill_Style);
-  flags.setColor(cp->GetColor(kColorBraveVerticalTabSeparator));
-  canvas->DrawRect(gfx::RectF(separator_bounds), flags);
+  OnPaintFill(canvas);
 }
 
-void BraveNewTabButton::FrameColorsChanged() {
-  NewTabButton::FrameColorsChanged();
-  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs))
-    return;
-
-  DCHECK(text_ && shortcut_text_);
-  auto text_color = GetForegroundColor();
-  text_->SetEnabledColor(text_color);
-  shortcut_text_->SetEnabledColor(text_color);
+void BraveNewTabButton::OnPaintFill(gfx::Canvas* canvas) const {
+  NewTabButton::PaintFill(canvas);
 }
 
-void BraveNewTabButton::Layout() {
-  NewTabButton::Layout();
-  if (!base::FeatureList::IsEnabled(tabs::features::kBraveVerticalTabs))
-    return;
-
-  DCHECK(text_ && shortcut_text_);
-  const bool show_vertical_tabs =
-      tabs::utils::ShouldShowVerticalTabs(tab_strip_->GetBrowser());
-  text_->SetVisible(show_vertical_tabs);
-  shortcut_text_->SetVisible(show_vertical_tabs);
-  if (!text_->GetVisible())
-    return;
-
-  constexpr int kTextLeftMargin = 20 + kInsetForVerticalTab;
-  gfx::Rect text_bounds = GetLocalBounds();
-  text_bounds.Inset(
-      gfx::Insets().set_left(kTextLeftMargin).set_right(kInsetForVerticalTab));
-  text_->SetBoundsRect(text_bounds);
-  shortcut_text_->SetBoundsRect(text_bounds);
+gfx::Insets BraveNewTabButton::GetInsets() const {
+  // Give an additional left margin to make more space from tab.
+  // TabStripRegionView::UpdateNewTabButtonBorder() gives this button's inset.
+  // So, adding more insets here is easy solution.
+  return NewTabButton::GetInsets() + gfx::Insets::TLBR(0, 6, 0, 0);
```

Comment:
Is this the same `6` that is used [here](https://github.com/brave/brave-core/pull/17732/files#diff-5b35e9275153a15ac587b50c869187b3def209db671eb7c8e6cbbc2b6512fcafR263)? Maybe it's worth putting them in a variable?

Reply:
Sorry, the link doesn't seem to lead me to the right place. Could you give me another pointer?

Reply:
Hmm, that's weird - I think the Github link is broken. When I tried getting it again it gave me the same link and it definitely doesn't take you to the right place :laughing: 

Does this one work?
https://github.dev/brave/brave-core/blob/faf5ed87c5429c3a7ab7fc9c453f637cf2d3ea42/browser/ui/views/frame/vertical_tab_strip_region_view.cc#L263

Reply:
Awesome! Unfortunately, that inset is a coincidence.